File: backend/app/routes/security.py
from fastapi import APIRouter, Depends, HTTPException
from pydantic import BaseModel

from app.models.user import User
from app.security.dependencies import get_current_user
from app.services import security_service

router = APIRouter(prefix="/api/projects", tags=["security"])


class SecurityFindingOut(BaseModel):
    id: str
    tool: str
    severity: str
    file: str | None
    line: int | None
    issue: str
    evidence: str | None
    recommendation: str | None


class ScanSummaryOut(BaseModel):
    scan_id: str
    provider_status: dict[str, str]
    total_findings: int


@router.post("/{project_id}/security-scans", response_model=ScanSummaryOut)
async def run_scan(project_id: str, user: User = Depends(get_current_user)):
    try:
        summary = await security_service.run_security_scan(user.organization_id, project_id)
    except security_service.ProjectNotConnected as exc:
        raise HTTPException(status_code=422, detail=str(exc))
    return ScanSummaryOut(**summary)


@router.get("/{project_id}/security-findings", response_model=list[SecurityFindingOut])
async def list_findings(project_id: str, user: User = Depends(get_current_user)):
    findings = await security_service.list_findings(user.organization_id, project_id)
    return [SecurityFindingOut(**f.model_dump()) for f in findings]

File: frontend/src/services/api/security.ts
import { apiClient } from "./client";

export interface SecurityFindingOut {
  id: string;
  tool: string;
  severity: string;
  file: string | null;
  line: number | null;
  issue: string;
  evidence: string | null;
  recommendation: string | null;
}

export interface ScanSummaryOut {
  scan_id: string;
  provider_status: Record<string, string>;
  total_findings: number;
}

export async function runSecurityScan(projectId: string): Promise<ScanSummaryOut> {
  const { data } = await apiClient.post<ScanSummaryOut>(`/api/projects/${projectId}/security-scans`);
  return data;
}

export async function listSecurityFindings(projectId: string): Promise<SecurityFindingOut[]> {
  const { data } = await apiClient.get<SecurityFindingOut[]>(`/api/projects/${projectId}/security-findings`);
  return data;
}
